### PR TITLE
[SPARK-18447][DOCS] Fix the markdown for `Note:`/`NOTE:`/`Note that` across Python API documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

It seems in Python, there are

- `Note:`
- `NOTE:`
- `Note that`
- `.. note::`

This PR proposes to fix those to `.. note::` to be consistent.

**Before**

**After**

## How was this patch tested?

The notes were found via

```bash 
grep -r "Note: " .
grep -r "NOTE: " .
grep -r "Note that " .
```

And then fixed one by one comparing with API documentation.

After that, manually tested via `make html` under `./python/docs`.

### CODE REVIEW

File: python/pyspark/context.py [hunk 1]
```diff
@@ -520,8 +520,8 @@ def wholeTextFiles(self, path, minPartitions=None, use_unicode=True):
           ...
           (a-hdfs-path/part-nnnnn, its content)
 
-        NOTE: Small files are preferred, as each file will be loaded
-        fully in memory.
+        .. note:: Small files are preferred, as each file will be loaded
+            fully in memory.
```
Comment (HyukjinKwon) [inline python/pyspark/context.py]: (Note with doctests below)

File: python/pyspark/rdd.py [hunk 2]
```diff
@@ -1220,10 +1219,10 @@ def top(self, num, key=None):
         """
         Get the top N elements from an RDD.
 
-        Note that this method should only be used if the resulting array is expected
-        to be small, as all the data is loaded into the driver's memory.
+        .. note:: This method should only be used if the resulting array is expected
+            to be small, as all the data is loaded into the driver's memory.
 
-        Note: It returns the list sorted in descending order.
+        .. note:: It returns the list sorted in descending order.
```
Comment (HyukjinKwon) [inline python/pyspark/rdd.py]: (Multiple notes)

File: python/pyspark/rdd.py [hunk 3]
```diff
@@ -386,10 +386,8 @@ def sample(self, withReplacement, fraction, seed=None):
             with replacement: expected number of times each element is chosen; fraction must be >= 0
         :param seed: seed for the random number generator
 
-        .. note::
-
-            This is not guaranteed to provide exactly the fraction specified of the total count
-            of the given :class:`DataFrame`.
+        .. note:: This is not guaranteed to provide exactly the fraction specified of the total
+            count of the given :class:`DataFrame`.
```
Comment (HyukjinKwon) [inline python/pyspark/rdd.py]: This is actually already fine but I just made the indentation consistent with the others.

**Comment by HyukjinKwon** [comment]
cc @srowen and @aditya1702 who was interested in this issue.

File: python/pyspark/ml/linalg/__init__.py [hunk 4]
```diff
@@ -746,11 +746,12 @@ def __hash__(self):
...
 
     """
-    Factory methods for working with vectors. Note that dense vectors
-    are simply represented as NumPy array objects, so there is no need
-    to covert them for use in MLlib. For sparse vectors, the factory
-    methods in this class create an MLlib-compatible type, or users
-    can pass in SciPy's C{scipy.sparse} column vectors.
+    Factory methods for working with vectors.
+
+    .. note:: Dense vectors are simply represented as NumPy array objects,
```
Comment (srowen) [inline python/pyspark/ml/linalg/__init__.py]: This is another example where I think it works either way -- it isn't necessarily a stand-alone note, as much as a sentence that begins "Note that...", but it works OK either way in these cases.

**Review by srowen — APPROVED** [review]

**Comment by jagadeesanas2** [comment]
LGTM

**Comment by srowen** [comment]
Merged to master/2.1

**Pull request closed without merging**